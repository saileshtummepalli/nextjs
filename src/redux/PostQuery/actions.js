export const postActionTypes = {
  FAILURE: "FAILURE",
  CREATE_DATA: "CREATE_DATA",
  CREATE_DATA_SUCCESS: "CREATE_DATA_SUCCESS",
  HYDRATE: "HYDRATE",
};

export function failure(error) {
  return {
    type: postActionTypes.FAILURE,
    error,
  };
}

export const CreateData = ({ data }) => {
  return {
    type: postActionTypes.CREATE_DATA,
    data,
  };
};

export const createDataSaga = ({ data }) => {
  return {
    type: postActionTypes.CREATE_DATA_SUCCESS,
    data,
  };
};
