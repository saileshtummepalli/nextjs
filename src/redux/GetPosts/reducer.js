import { actionTypes } from "./actions";
import { HYDRATE } from "next-redux-wrapper";

const initialState = {
  error: false,
  placeholderData: null,
  isLoading:true
};

function reducer(state =initialState, action) {
  console.log(action,"action")
  switch (action.type) {
    case HYDRATE:
      return { ...state, ...action.payload };
    case actionTypes.FAILURE:
      return {
        ...state,
        ...{ error: action.error  ,isLoading:false},
      };
      case actionTypes.LOAD_DATA:
      return {
        ...state,
        ...{ isLoading: true },
      };
    case actionTypes.LOAD_DATA_SUCCESS:
      return {
        ...state,
         placeholderData: action.data ,isLoading:false ,
      };

    default:
      return state;
  }
}

export default reducer;
