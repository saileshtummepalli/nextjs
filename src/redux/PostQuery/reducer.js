import { postActionTypes } from "./actions";
import { HYDRATE } from "next-redux-wrapper";

const initialState = {
  error: null,
  data: {},
};
const postReducer = (state = initialState, action) => {
  switch (action.type) {
    case HYDRATE:
      return { ...state, ...action.payload };

    case action.CREATE_DATA:
      return {
        ...state,
        error: null,
      };
    case postActionTypes.CREATE_DATA_SUCCESS:
      return {
        ...state,
        ...{ data: action.data, error: null },
      };
    case postActionTypes.FAILURE:
      return {
        ...state,
        ...{ error: action.error },
      };
    default:
      return state;
  }
};

export default postReducer;
