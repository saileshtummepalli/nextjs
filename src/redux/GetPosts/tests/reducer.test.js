import reducer, { initialState } from "../reducer";
import {actionTypes} from "../actions"

describe("GetPosts", () => {
it('Load data on successfull fetch ',() =>{
  const mockdata= "test"
  expect(reducer(initialState,{
    type:actionTypes.LOAD_DATA_SUCCESS,
    data:mockdata
  })).toEqual({
    placeholderData:mockdata,
    error: false,
    isLoading: false
  });
});
});