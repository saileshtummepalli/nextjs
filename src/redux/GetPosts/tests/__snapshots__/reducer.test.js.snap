// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetPosts reducer inital state 1`] = `
Object {
  "error": false,
  "isLoading": true,
  "placeholderData": null,
}
`;
